daemon/src/db/cull_old_dlcs.rs

```diff
@@ -0,0 +1,307 @@
+    use crate::db::memory;
+    use bdk::bitcoin::Amount;
+    use model::libp2p::PeerId;
+    use model::Cfd;
+    use model::CfdEvent;
+    use model::EventKind;
+    use model::FundingRate;
+    use model::Leverage;
+    use model::OpeningFee;
+    use model::OrderId;
+    use model::Position;
+    use model::Price;
+    use model::Role;
+    use model::Timestamp;
+    use model::TxFeeRate;
+    use model::Usd;
+    use rust_decimal_macros::dec;
+    use sqlx::Connection;
+    use std::num::NonZeroU8;
+    use time::Duration;
+
+    #[tokio::test]
+    async fn given_two_rollovers_when_cull_old_dlcs_then_only_last_one_has_some_dlc() {
+        let db = memory().await.unwrap();
+
+        let (cfd, contract_setup, rollovers) = rolled_over_cfd(NonZeroU8::new(2).unwrap());
+
+        db.insert_cfd(&cfd).await.unwrap();
+
+        db.append_event(contract_setup).await.unwrap();
+
+        for rollover in rollovers {
+            db.append_event(rollover).await.unwrap();
+        }
+
+        db.cull_old_dlcs().await.unwrap();
+
+        let mut conn = db.inner.acquire().await.unwrap();
+        let mut db_tx = conn.begin().await.unwrap();
+        let events = load_cfd_events(&mut db_tx, cfd.id(), 0).await.unwrap();
+        db_tx.commit().await.unwrap();
+
+        assert!(matches!(
+            &events[0].event,
+            EventKind::ContractSetupCompleted { dlc: None, .. },
+        ));
+
+        assert!(matches!(
+            &events[1].event,
+            EventKind::RolloverCompleted { dlc: None, .. },
+        ));
+
+        assert!(matches!(
+            &events[2].event,
+            EventKind::RolloverCompleted { dlc: Some(_), .. },
+        ));
+    }
+
+    #[tokio::test]
+    async fn given_three_rollovers_when_cull_old_dlcs_then_only_last_one_has_some_dlc() {
+        let db = memory().await.unwrap();
+
+        let (cfd, contract_setup, rollovers) = rolled_over_cfd(NonZeroU8::new(3).unwrap());
+
+        db.insert_cfd(&cfd).await.unwrap();
+
+        db.append_event(contract_setup).await.unwrap();
+
+        for rollover in rollovers {
+            db.append_event(rollover).await.unwrap();
+        }
+
+        db.cull_old_dlcs().await.unwrap();
+
+        let mut conn = db.inner.acquire().await.unwrap();
+        let mut db_tx = conn.begin().await.unwrap();
+        let events = load_cfd_events(&mut db_tx, cfd.id(), 0).await.unwrap();
+        db_tx.commit().await.unwrap();
+
+        assert!(matches!(
+            &events[0].event,
+            EventKind::ContractSetupCompleted { dlc: None, .. },
+        ));
+
+        assert!(matches!(
+            &events[1].event,
+            EventKind::RolloverCompleted { dlc: None, .. },
+        ));
+
+        assert!(matches!(
+            &events[2].event,
+            EventKind::RolloverCompleted { dlc: None, .. },
+        ));
+
+        assert!(matches!(
+            &events[3].event,
+            EventKind::RolloverCompleted { dlc: Some(_), .. },
```

Comment:
🙃 I think it would be better to compare against the actual expected DLC - or at least something that makes sure that this is the DLC we expect (as in the latest one) and not e.g. the one of contract setup. Given the logic this should always work out, but still.